src/index.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  IAssignmentLogger,
+  validation,
+  IEppoClient,
+  EppoClient,
+  FlagConfigurationRequestParameters,
+  MemoryOnlyConfigurationStore,
+  Flag,
+  HybridConfigurationStore,
+  MemoryStore,
+} from '@eppo/js-client-sdk-common';
+
+import { sdkName, sdkVersion } from './sdk-data';
+import VercelAsyncStore from './vercel-async-store';
+
+export {
+  IAssignmentLogger,
+  IAssignmentEvent,
+  IEppoClient,
+  IAsyncStore,
+} from '@eppo/js-client-sdk-common';
+
+/**
+ * Configuration used for initializing the Eppo client
+ * @public
+ */
+export interface IClientConfig {
+  /**
+   * Eppo API key
+   */
+  apiKey: string;
+
+  /**
+   * Params for setting up connection with Vercel
+   *
+   * edgeConfig - string given on creation of Vercel Edge Config Store.
+   * edgeConfigStoreId - id of created Vercel Edge Config Store.
+   * vercelApiToken - Vercel API token with write access. Needed to write configs to Vercel Edge Config Store.
+   */
+  vercelParams: {
+    edgeConfig: string;
+    edgeConfigStoreId: string;
+    vercelApiToken: string;
+  };
+
+  /**
+   * Base URL of the Eppo API.
+   * Clients should use the default setting in most cases.
+   */
+  baseUrl?: string;
+
+  /**
+   * Pass a logging implementation to send variation assignments to your data warehouse.
+   */
+  assignmentLogger: IAssignmentLogger;
+
+  /***
+   * Timeout in milliseconds for the HTTPS request for the experiment configuration. (Default: 5000)
+   */
+  requestTimeoutMs?: number;
+
+  /**
+   * Number of additional times the initial configuration request will be attempted if it fails.
+   * This is the request typically synchronously waited (via await) for completion. A small wait will be
+   * done between requests. (Default: 1)
+   */
+  numInitialRequestRetries?: number;
+
+  /**
+   * Throw an error if unable to fetch an initial configuration during initialization. (default: true)
+   */
+  throwOnFailedInitialization?: boolean;
+}
```

Comment:
I'd expect another optional config option of how long until we consider the cache expired (instead of hard-coded 20 seconds)